.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,49 @@
+name: Unit tests
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+    branches:
+      - main
+      - stable-*
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    name: Unit tests (Ⓐ${{ matrix.ansible }}+py${{ matrix.python }})
+    strategy:
+      fail-fast: false
+      matrix:
+        ansible:
+          - stable-2.14
+          - stable-2.15
+          - stable-2.16
+          - milestone
+          - devel
+        python:
+          - '3.9'
+          - '3.10'
+          - '3.11'
+          - '3.12'
+        exclude:
+          - ansible: stable-2.14
+            python: '3.12'
+          - ansible: stable-2.15
+            python: '3.12'
+          - ansible: stable-2.16
+            python: '3.9'
+          - ansible: milestone
+            python: '3.9'
+          - ansible: devel
+            python: '3.9'
+    continue-on-error: ${{ matrix.ansible == 'devel' }}
+    steps:
+      - name: Perform unit testing
+        uses: ansible-community/ansible-test-gh-action@release/v1
```

Comment:
why not using standard workflow like others collections ? `ansible-network/github_actions/.github/workflows/unit_source.yml@main` or update directly the workflow from the ansible-network/github_actions in order for others collections to benefit from it

Reply:
I tried...but this collection didn't play nicely with pytest's test discovery and it generated a bunch of errors. Rather than try to resolve those now I figured we could continue using the ansible-test action since this repo was already using it, and change it to the ansible-network workflow/action when we fully standardize the collections.